### PR TITLE
[ARIES-1573] Configuration from properties

### PR DESCRIPTION
**Pull request by dpishchukhin**

### CODE REVIEW

File: discovery/config/src/main/java/org.apache.aries.rsa.discovery.config/Activator.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+package org.apache.aries.rsa.discovery.config;
+
+import org.osgi.framework.*;
+import org.osgi.service.cm.ManagedServiceFactory;
+import org.osgi.service.remoteserviceadmin.EndpointListener;
+import org.osgi.util.tracker.ServiceTracker;
+
+import java.util.Hashtable;
+
+@SuppressWarnings("deprecation")
```
Comment (cschneider) [inline discovery/config/src/main/java/org.apache.aries.rsa.discovery.config/Activator.java]: This does not seem to be necessary

File: discovery/config/src/main/java/org.apache.aries.rsa.discovery.config/ConfigDiscovery.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedServiceFactory;
+import org.osgi.service.remoteserviceadmin.EndpointDescription;
+import org.osgi.service.remoteserviceadmin.EndpointListener;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+@SuppressWarnings("deprecation")
```
Comment (cschneider) [inline discovery/config/src/main/java/org.apache.aries.rsa.discovery.config/ConfigDiscovery.java]: Does not seem to show deprecation warning on my side

File: discovery/config/src/main/java/org.apache.aries.rsa.discovery.config/Activator.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    private ServiceTracker<EndpointListener, EndpointListener> listenerTracker;
+    private ConfigDiscovery configDiscovery;
+    private ServiceRegistration<ManagedServiceFactory> registration;
+
+    public void start(BundleContext context) {
+        configDiscovery = new ConfigDiscovery();
+        listenerTracker = new EPListenerTracker(context, configDiscovery);
+        listenerTracker.open();
+        registration = context.registerService(ManagedServiceFactory.class, configDiscovery, new Hashtable<String, Object>() {{
```
Comment (cschneider) [inline discovery/config/src/main/java/org.apache.aries.rsa.discovery.config/Activator.java]: Better not use the static put style. It creates a anon subclass and spits a serial warning.

**Comment by dpishchukhin** [comment]
changes were applied.

**Comment by cschneider** [comment]
You also need to change the directory "org.apache.aries...:" to hierarchical subdirs. ord/apache/aries/...

and please squash into one commit again.

**Comment by dpishchukhin** [comment]
I did not get you comment regarding subdirs. Can you explain, pls?

**Comment by cschneider** [comment]
If you look into src/main/java you see a single directory "org.apache.aries.rsa.discovery.config". Each package should be a separate directory though. I guess you use intellij which might not complain about this but eclipse does.

**Comment by dpishchukhin** [comment]
done. hope, now is fine.

tnx for your support

**Pull request closed without merging**